DataBiosphere/leonardo
# IA-1591 Add GCE+Dataproc algebra to Back Leo

Most of the meat of GCE is in here. High level changs:

* Removed `ClusterHelper`, `ComputeHelper` -- created `RuntimeAlgebra`, `DataprocInterpreter`, `GceInterpreter` which is the main integration point with Google/wb-libs.
* Made `ClusterMonitorActor`/`ClusterMonitorSupervisor` handle both GCE and Dataproc
* Made `LeoPubsubMessageSubscriber` handle both GCE and Dataproc
* Made `BucketHelper`, `VPCHelper` take an `F[_]` type parameter instead of `IO`
* Some pubsub model refactoring/renaming

TODO:
- ~Test the actual GCE creation~
- Add routes for GCE stop/start/delete/etc
- Zombie monitor is still Dataproc specific
- Error handling -- implement GCE getErrorDetails() and clean up on failure
   - In fact GCE doesn't fail if the init script fails, how to signal errors?
- Create dedicated VPC subnet instead of using default
- Merge wb-libs PR once all is working & tested
- Improve unit tests around moved/new files
- (Eventually) add Swagger page for new routes

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:


In all cases:
